### PR TITLE
DB/Quest: Head of Nefarian (A)

### PR DESCRIPTION
**Pull request by poszer**
- Head of Nefarian correctly displayed in Stormwind after quest turn-in to Field Marshal Afrasiabi.

Alliance [Quest](https://wowgaming.altervista.org/aowow/?quest=7782)

##### CHANGES PROPOSED:

-  SAI Update
-  

###### ISSUES ADDRESSED:

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/1661

##### TESTS PERFORMED:

Tested in game, quest can be completed. The Gameobject at the quest finish is correctly spawned.

##### HOW TO TEST THE CHANGES:

1. additem 19003 to an alliance character lvl60+.
2. accept quest and follow it through.
3. event plays out, players get buff, but no dragon head hanging.
4. [Image here](http://prntscr.com/n5sude)

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):

Master

### CODE REVIEW

**Comment by Pondaveia** [comment]
I tested:
1. The head shows a tooltip and I doubt it should
2. The buff doesn't work

**Comment by poszer** [comment]
> I tested:
> 
> 1. The head shows a tooltip and I doubt it should
> 2. The buff doesn't work
> 3. When should it despawn?

What buff should be applied ? I don't see anything related on wow db about that 

> Upon completion of this quest you will gain:
> 14,300 experience
> 500 reputation with Stormwind
> 500 reputation with Ironforge
> 500 reputation with Darnassus
> 500 reputation with Gnomeregan Exiles
> 500 reputation with Exodar

**Comment by comix1988** [comment]
i found this @poszer 
https://wow.gamepedia.com/Rallying_Cry_of_the_Dragonslayer

**Comment by poszer** [comment]
> i found this @poszer
> https://wow.gamepedia.com/Rallying_Cry_of_the_Dragonslayer

That spell is already on SAI actionlist for cast. Will check now why doesn't work

**Comment by poszer** [comment]
Also how did you try it, with level 80 ? If yes, get the level 60 / 60+ and try it
level60

**Comment by comix1988** [comment]
- tested at level 60 and I get the buff
- tested at level 80 and buff is not applied

good job @poszer

**Comment by poszer** [comment]
> * tested at level 60 and I get the buff
> * tested at level 80 and buff is not applied
> 
> but when should it despawn?
> 
> good job @poszer

Will take a look on despawn now, tryed to find timing, but nothing related. Anyone of you know despawn time ? @Pondaveia @comix1988

File: data/sql/updates/pending_db_world/rev_1554113581920680700.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@ENTRY,9,0,0,0,0,100,0,1000,1000,0,0,83,2,0,0,0,0,0,1,0,0,0,0,0,0,0,'Field Marshal Afrasiabi - On Script - Remove Npc Flag Questgiver'),
+(@ENTRY,9,1,0,0,0,100,0,5000,5000,0,0,1,0,8000,0,0,0,0,12,1,0,0,0,0,0,0,'Field Marshal Afrasiabi - On Script - Say Line 0'),
+(@ENTRY,9,2,0,0,0,100,0,8000,8000,0,0,1,1,10000,0,0,0,0,12,1,0,0,0,0,0,0,'Field Marshal Afrasiabi - On Script - Say Line 1'),
+(@ENTRY,9,3,0,0,0,100,0,3000,3000,0,0,50,179882,21600,0,0,0,0,8,0,0,0,-8925.57,496.042,103.767,2.42801,"Field Marshal Afrasiabi - On Script - 'Summon Gameobject'The Severed Head of Nefarian"),
+(@ENTRY,9,4,0,0,0,100,0,6000,6000,0,0,11,22888,0,0,0,0,0,1,0,0,0,0,0,0,0,"Field Marshal Afrasiabi - On Script - Cast 'Rallying Cry of the Dragonslayer'"),
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1554113581920680700.sql]: Please, change " to ' to keep consistency.

File: data/sql/updates/pending_db_world/rev_1554113581920680700.sql [hunk 2]
```diff
@@ -15,8 +15,8 @@ INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type
 (@ENTRY,9,0,0,0,0,100,0,1000,1000,0,0,83,2,0,0,0,0,0,1,0,0,0,0,0,0,0,'Field Marshal Afrasiabi - On Script - Remove Npc Flag Questgiver'),
 (@ENTRY,9,1,0,0,0,100,0,5000,5000,0,0,1,0,8000,0,0,0,0,12,1,0,0,0,0,0,0,'Field Marshal Afrasiabi - On Script - Say Line 0'),
 (@ENTRY,9,2,0,0,0,100,0,8000,8000,0,0,1,1,10000,0,0,0,0,12,1,0,0,0,0,0,0,'Field Marshal Afrasiabi - On Script - Say Line 1'),
-(@ENTRY,9,3,0,0,0,100,0,3000,3000,0,0,50,179882,21600,0,0,0,0,8,0,0,0,-8925.57,496.042,103.767,2.42801,"Field Marshal Afrasiabi - On Script - 'Summon Gameobject'The Severed Head of Nefarian"),
-(@ENTRY,9,4,0,0,0,100,0,6000,6000,0,0,11,22888,0,0,0,0,0,1,0,0,0,0,0,0,0,"Field Marshal Afrasiabi - On Script - Cast 'Rallying Cry of the Dragonslayer'"),
+(@ENTRY,9,3,0,0,0,100,0,3000,3000,0,0,50,179882,21600,0,0,0,0,8,0,0,0,-8925.57,496.042,103.767,2.42801,'Field Marshal Afrasiabi - On Script - Summon Gameobject The Severed Head of Nefarian'),
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1554113581920680700.sql]: To keep the ' just put a \ in front of them :)

**Comment by gtao725** [comment]
@poszer From my memory it was 2 hours for the head to show in either Stormwind or Orgrimmar depending on the faction in question. Because the buff (lvl 63 and below) you get only lasts 120min. This makes sense as it then enables other players to hand in the same quest after 2 hours has passed. Pls also check the despawn time of the head for the horde side, I have not really timed the actual despawn time.

**Comment by Pondaveia** [comment]
This is strange, other players won't be able to complete the quest until the head despawns or get the buff? Shouldn't the head despawn whenever a player completes the quest and also give them the buff?

**Comment by poszer** [comment]
> This is strange, other players won't be able to complete the quest until the head despawns or get the buff? Shouldn't the head despawn whenever a player completes the quest and also give them the buff?

Can't find a lot infos about this -.- We need to know despawn time, and how this quest can be completed if more people have same quest accepted

**Comment by Pondaveia** [comment]
Maybe someone has some sniffs that can share, would be nice. @BarbzYHOOL You are from the underground, maybe you can pull the right strings?

**Comment by BarbzYHOOL** [comment]
Read the comments here, it's same mechanism

https://www.wowhead.com/item=18423/head-of-onyxia#comments

**Comment by Pondaveia** [comment]
It confirms my theory about despawning the existing one and spawning a new... That and onixya giving epic head

**Comment by gtao725** [comment]
> This is strange, other players won't be able to complete the quest until the head despawns or get the buff? Shouldn't the head despawn whenever a player completes the quest and also give them the buff?

@Pondaveia Sorry I probably didn't describe what I wanted to say clearly. I remember the buff to be 2 hrs and so is the NATURAL despawn time of the head (hanging in Stormwind or on a pike in Orgrimmar).  If another player hands in the quest within the 2 hour period, then my understanding is that the buff gets refreshed AND the head hanging/head on a pike despawn time also resets back to 2 hours. So yes of course any player is able to hand in the quest anytime. Hope this clears up the confusion.

**Comment by gtao725** [comment]
> It confirms my theory about despawning the existing one and spawning a new... That and onixya giving epic head

Regarding Onyxia, since WotLK Patch 3.2.2, the original level 60 raid encounter has being replaced with a level 80 raid encounter. With this change, Blizzard also REMOVED the original quest chain (https://wowgaming.altervista.org/aowow/?quest=7490 for horde and https://wowgaming.altervista.org/aowow/?quest=7495 for alliance). The removed quests were then replaced with https://wowgaming.altervista.org/aowow/?quest=24429 for horde and https://wowgaming.altervista.org/aowow/?quest=24428 for alliance. What this means is that Onyxia's Head will never again be shown in either Stormwind or Orgrimmar since the quest is now turned in to different people and there are no event triggers associated with the new quest turn-ins to hang or mount the head in the capital cities. The buff (Rallying Cry of the Dragonslayer) is also NOT GIVEN ANYMORE for turning in Level 80 Onyxia's Head.

**Comment by BarbzYHOOL** [comment]
imo what Gtao suggests is better than the comment on wowhead, more logical

**Comment by gtao725** [comment]
I am just wondering has anyone tested how the current working Horde version of this quest's despawn time and condition works? As a matter of fact I am going to test that now myself.

**Comment by Pondaveia** [comment]
But we all want to see Onixya hanging by a spike :( But it makes sence since they released the remaked Onixya raid and so there for the quest got updated. Good research @gtao725

**Comment by gtao725** [comment]
Ok, For the Head of Nefarian in Orgrimmar, 2 hrs passed and the head is still there on a spike. No idea how long it stays, could someone maybe check the script? I can confirm that if someone turns the quests in after somebody else turns it in the buff gets refreshed.

**Comment by BarbzYHOOL** [comment]
from my faulty memory, i think the head stayed more than 2 hours in vanilla like 24hrs

**Comment by poszer** [comment]
> from my faulty memory, i think the head stayed more than 2 hours in vanilla like 24hrs

Good, will edit to 2h that :smile:

**Comment by BarbzYHOOL** [comment]
But it's not proven, but since I saw the head multiple times, it's logic that it stays more than 2 hours else I would've rarely seen it.

How is it on vanilla cores?

**Comment by gtao725** [comment]
maybe this will help:

https://github.com/TrinityCore/TrinityCore/issues/12256

**Comment by poszer** [comment]
> maybe this will help:
> 
> [TrinityCore/TrinityCore#12256](https://github.com/TrinityCore/TrinityCore/issues/12256)

Nah, they don't have despawn there. We will use 2 hours, when someone found the correct despawn time, it's easy to update it :smile:

**Comment by gtao725** [comment]
> > maybe this will help:
> > [TrinityCore/TrinityCore#12256](https://github.com/TrinityCore/TrinityCore/issues/12256)
> 
> Nah, they don't have despawn there. We will use 2 hours, when someone found the correct despawn time, it's easy to update it

ok. What is the current despawn time for horde version? Perhaps also make it consistent by changing it to 2 hrs until someone can confirm a definitive time?

**Comment by poszer** [comment]
> > > maybe this will help:
> > > [TrinityCore/TrinityCore#12256](https://github.com/TrinityCore/TrinityCore/issues/12256)
> > 
> > 
> > Nah, they don't have despawn there. We will use 2 hours, when someone found the correct despawn time, it's easy to update it
> 
> ok. What is the current despawn time for horde version? Perhaps also make it consistent by changing it to 2 hrs until someone can confirm a definitive time?

Yeah, we will use for horde 2 hours too. Thanks for all infos :+1:

**Comment by gtao725** [comment]
no probs I will keep searching and update once I find something useful.

**Comment by gtao725** [comment]
Just confirmed the current behavior of the horde version of this quest in Azerothcore. The head on spike remains forever until the server is restarted. To be honest I think displaying the head for 2 hours is more logical, but I can also live with it being displayed until server restart. Either way is fine. On official Blizzard servers, there is the weekly maintenance, so if the head does get shown indefinitely until server restart, the display period at most will be a week.

**Comment by poszer** [comment]
> Just confirmed the current behavior of the horde version of this quest in Azerothcore. The head on spike remains forever until the server is restarted. To be honest I think displaying the head for 2 hours is more logical, but I can also live with it being displayed until server restart. Either way is fine. On official Blizzard servers, there is the weekly maintenance, so if the head does get shown indefinitely until server restart, the display period at most will be a week.

Done, both are set to despawn in 2h https://github.com/azerothcore/azerothcore-wotlk/pull/1686

**Comment by gtao725** [comment]
@poszer Sweet! Thx for the hard work, can merge soon hopefully.

**Comment by BarbzYHOOL** [comment]
Hmm interesting fact about maximum duration = a week

**Comment by gtao725** [comment]
I am actually good with it either way, be it 2 hrs - 7 days, as long as the quest works correctly, from solely a player's point of view lol.

**Pull request merged**